src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFile2.cs

```diff
@@ -2,102 +2,165 @@
+			{
+				Assert.Fail("DateCreated: too early - method doesnt work");
+			}
+
+			// second, date should not be in future
+			if (dateOnCreating > dateAfterCreating.AddSeconds(2))
+			{
+				Assert.Fail("DateCreated: too late - method doesnt work");
+			}
+
+			// third, it should not be "datetime.now"
+			var initialTimeDiff = DateTimeOffset.Now - dateOnCreating;
+			int loopGuard;
+			for (loopGuard = 20; loopGuard > 0; loopGuard--) // wait for date change for max 5 seconds
 			{
-                Assert.Fail("DateCreated: too early - method doesnt work");
+				await Task.Delay(250);
+				dateOnCreating = testFile.DateCreated;
+				if (DateTimeOffset.Now - dateOnCreating > initialTimeDiff)
+				{
+					break;
+				}
 			}
-
-			// second, date should not be in future
-			if(dateOnCreating > dateAfterCreating.AddSeconds(2))
+			if (loopGuard < 1)
+			{
+				Assert.Fail("DateCreated: probably == DateTime.Now, - method doesnt work");
+			}
+
+
+
+			// now, test Date Modified
+
+			DateTimeOffset dateBeforeModify = DateTimeOffset.Now;
+			// modify file
+			Stream fileStream = await testFile.OpenStreamForWriteAsync();
+			fileStream.Seek(0, SeekOrigin.End);
+			StreamWriter streamWriter = new StreamWriter(fileStream);
+			streamWriter.Write(_filename);  // anything - just write String
+			streamWriter.Dispose();
+			fileStream.Dispose();
+			DateTimeOffset dateAfterModify = DateTimeOffset.Now;
+
+			// get DateModified
+			DateTimeOffset dateModified = DateTimeOffset.Now; // unneeded initialization - just to skip compiler error of using uninitialized variable
+			try
 			{
-                Assert.Fail("DateCreated: too late - method doesnt work");
+				dateModified = (await testFile.GetBasicPropertiesAsync()).DateModified;
 			}
-
-			// third, it should not be "datetime.now"
-			var initialTimeDiff = DateTimeOffset.Now - dateOnCreating;
-			int loopGuard; 
-			for(loopGuard = 20; loopGuard > 0; loopGuard--) // wait for date change for max 5 seconds
+			catch
+			{
+				Assert.Fail("DateModified exception - error in tested method");
+			}
+
+			// first, date should not be year 1601 or something like that...
+			if (dateModified < dateBeforeModify.AddSeconds(-2))
+			{
+				Assert.Fail("DateModified: too early - method doesnt work");
+			}
+
+			// second, date should not be in future
+			if (dateModified > dateAfterModify.AddSeconds(2))
+			{
+				Assert.Fail("DateCreated: too late - method doesnt work");
+			}
+
+			// third, it should not be "datetime.now"
+			initialTimeDiff = DateTimeOffset.Now - dateModified;
+			for (loopGuard = 20; loopGuard > 0; loopGuard--) // wait for date change for max 5 seconds
 			{
 				await Task.Delay(250);
-				dateOnCreating = testFile.DateCreated;
-				if(DateTimeOffset.Now - dateOnCreating > initialTimeDiff)
+				var dateModifiedLoop = (await testFile.GetBasicPropertiesAsync()).DateModified;
+				if (DateTimeOffset.Now - dateModifiedLoop > initialTimeDiff)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4941799)